flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -475,8 +475,10 @@ public void onBuffer(Buffer buffer, int sequenceNumber, int backlog) throws IOEx
                 DataType dataType = buffer.getDataType();
                 if (dataType.hasPriority()) {
                     firstPriorityEvent = addPriorityBuffer(sequenceBuffer);
+                    recycleBuffer = false;
```

Comment:
Could we move that outside of the `if-else` branch? I just see more issues arising this way.

Reply:
I was thinking about this, but such change would require an assumption that:
```
if (dataType.requiresAnnouncement()) {
    firstPriorityEvent = addPriorityBuffer(announce(sequenceBuffer));
}
```
Doesn't throw any exception ever. Which I think is not strictly speaking true.

Reply:
Okay fair point.